src/libopensc/card-openpgp.c

```diff
@@ -540,8 +563,15 @@ pgp_init(sc_card_t *card)
 				_sc_card_add_rsa_alg(card, 2048, flags_rsa, 0);
 				for (i=0; priv->ec_curves[i].oid.value[0] >= 0; i++)
 				{
-					_sc_card_add_ec_alg(card, priv->ec_curves[i].size,
-						flags_ecc, ext_flags, &priv->ec_curves[i].oid);
+					if (priv->ec_curves[i].key_type == SC_ALGORITHM_EC)
+						_sc_card_add_ec_alg(card, priv->ec_curves[i].size,
+								flags_ecc, ext_flags, &priv->ec_curves[i].oid);
+					else if (priv->ec_curves[i].key_type == SC_ALGORITHM_EDDSA)
+						_sc_card_add_eddsa_alg(card, priv->ec_curves[i].size,
+								flags_eddsa, ext_flags, &priv->ec_curves[i].oid);
+					else
```

Comment:
The `priv->ec_curve`  is a one of two static tables in the `card-openpgp.c`. If any new curve type is added  to the table, it will ben internal error.